### PR TITLE
Address residual URI todo

### PR DESCRIPTION
**Pull request by slavek-kucera**

### CODE REVIEW

File: utils/test/resource_location_test.cpp [hunk 1]
```diff
@@ -908,3 +908,37 @@ TEST(resource_location, replace_filename)
...
+
+TEST(resource_location, parent)
+{
+    EXPECT_EQ(resource_location("a").parent().get_uri(), "");
+    EXPECT_EQ(resource_location("x/a").parent().get_uri(), "x");
+    EXPECT_EQ(resource_location("x/").parent().get_uri(), "x");
+    EXPECT_EQ(resource_location("schema:a").parent().get_uri(), "schema:");
+    EXPECT_EQ(resource_location("schema://h/a").parent().get_uri(), "schema://h");
+    EXPECT_EQ(resource_location("schema://h/a/x/").parent().get_uri(), "schema://h/a/x");
+    EXPECT_EQ(resource_location("schema://h/a").parent().get_uri(), "schema://h");
```
Comment (jirimosinger) [inline utils/test/resource_location_test.cpp:932]: If I remember correctly, the `/` in URLs represents a directory. If the slash is missing then the path is supposed to lead to a file. 

Therefore I suggest:
```suggestion
    EXPECT_EQ(resource_location("schema://h/a").parent().get_uri(), "schema://h/");
    EXPECT_EQ(resource_location("schema://h/a/x/").parent().get_uri(), "schema://h/a/x/");
    EXPECT_EQ(resource_location("schema://h/a").parent().get_uri(), "schema://h/");
```

File: utils/include/utils/resource_location.h [hunk 2]
```diff
@@ -57,6 +57,11 @@ class resource_location
     resource_location& replace_filename(std::string_view other);
     static resource_location replace_filename(resource_location rl, std::string_view other);
 
+    std::string filename() const;
+    resource_location parent() const;
+
+    std::string get_local_if_available() const;
```
Comment (jirimosinger) [inline utils/include/utils/resource_location.h:63]: Hm, based on the function name I would expect it to return either a local path, empty string or `nullopt`.

I suggest one of the following:
1. returning an empty string or `nullopt` when local is not available
2. renaming to something like "get_local_or_non_local_uri", "get_local_or_uri"

File: utils/test/resource_location_test.cpp (hunk 1 above)
Comment (slavek-kucera) [reply utils/test/resource_location_test.cpp:932]: std::filesystem::path::parent_path() returns a path without the slash.
```
int main()
{
    std::cout << std::filesystem::path("/a/b/c/d").parent_path().string() << '\n';
    std::cout << std::filesystem::path("/a/b/c/").parent_path().string()  << '\n';
}
```
will print
```
/a/b/c
/a/b/c
```

Comment (jirimosinger) [reply utils/test/resource_location_test.cpp:932]: I agree that it's impractical but it would be RFC compliant. Let's discuss it in person

**Review by jirimosinger — APPROVED** [review]

**Pull request merged**